src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,758 @@
+
+    computeAllSeismoTraces( time_n, dt, ux_np1, ux_n, uxReceivers );
+    computeAllSeismoTraces( time_n, dt, uy_np1, uy_n, uyReceivers );
+    computeAllSeismoTraces( time_n, dt, uz_np1, uz_n, uzReceivers );
+
+    forAll< EXEC_POLICY >( nodeManager.size(), [=] GEOSX_HOST_DEVICE ( localIndex const a )
+    {
+      ux_nm1[a] = ux_n[a];
+      uy_nm1[a] = uy_n[a];
+      uz_nm1[a] = uz_n[a];
+      ux_n[a] = ux_np1[a];
+      uy_n[a] = uy_np1[a];
+      uz_n[a] = uz_np1[a];
+
+      stiffnessVectorx[a] = 0.0;
+      stiffnessVectory[a] = 0.0;
+      stiffnessVectorz[a] = 0.0;
+      rhsx[a] = 0.0;
+      rhsy[a] = 0.0;
+      rhsz[a] = 0.0;
+    } );
+
+  } );
+  return dt;
+
+}
+
+void ElasticWaveEquationSEM::cleanup( real64 const time_n, integer const, integer const, real64 const, DomainPartition & domain )
+{
+  // compute the remaining seismic traces, if needed
+  forMeshTargets( domain.getMeshBodies(), [&] ( string const &,
+                                                MeshLevel & mesh,
+                                                arrayView1d< string const > const & )
+  {
+    NodeManager & nodeManager = mesh.getNodeManager();
+    arrayView1d< real64 const > const ux_n = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementx_n >();
+    arrayView1d< real64 const > const ux_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementx_np1 >();
+    arrayView1d< real64 const > const uy_n = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementy_n >();
+    arrayView1d< real64 const > const uy_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementy_np1 >();
+    arrayView1d< real64 const > const uz_n = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementz_n >();
+    arrayView1d< real64 const > const uz_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Displacementz_np1 >();
+    arrayView2d< real64 > const uxReceivers   = m_displacementxNp1AtReceivers.toView();
+    arrayView2d< real64 > const uyReceivers   = m_displacementyNp1AtReceivers.toView();
+    arrayView2d< real64 > const uzReceivers   = m_displacementzNp1AtReceivers.toView();
+
+    computeAllSeismoTraces( time_n, 0, ux_np1, ux_n, uxReceivers );
+    computeAllSeismoTraces( time_n, 0, uy_np1, uy_n, uyReceivers );
+    computeAllSeismoTraces( time_n, 0, uz_np1, uz_n, uzReceivers );
+  } );
+}
+
+void ElasticWaveEquationSEM::computeAllSeismoTraces( real64 const time_n,
+                                                     real64 const dt,
+                                                     arrayView1d< real64 const > const var_np1,
+                                                     arrayView1d< real64 const > const var_n,
+                                                     arrayView2d< real64 > varAtReceivers )
+{
+  for( real64 timeSeismo;
+       (timeSeismo = m_dtSeismoTrace*m_indexSeismoTrace) <= (time_n + epsilonLoc) && m_indexSeismoTrace < m_nsamplesSeismoTrace;
+       m_indexSeismoTrace++ )
```

Comment:
```suggestion
  localIndex indexSeismoTrace = m_indexSeismoTrace;
  for( real64 timeSeismo;
       (timeSeismo = m_dtSeismoTrace*indexSeismoTrace) <= (time_n + epsilonLoc) && indexSeismoTrace < m_nsamplesSeismoTrace;
       indexSeismoTrace++ )
```

Reply:
Ok I see I'll fix it.

For the receivers and the stuff to move in WaveSolverBase I totally agree. I think for the stuff to put in WaveSolverBase it will be be a future PR at least for the acoustic case and maybe update this one directly (or change both at the same time it's something we need to discuss more then we have already done).
For the receiverts I think it is not reakky urgent because we generate then usually using PyGeosx and we get directly the array pnp1AtReceiver so we do not go threw the writing stuff inside the code (which in fact I think is devoted to disappear because it was mainly for debug purpose)